shared/libraries/packet_streaming/src/packet_streaming_client.cpp

```diff
@@ -74,23 +58,37 @@ void PacketStreamingClient::addEventPacketBuffer(const PacketBufferPtr& packetBu
 
     if (packet.getEventId() == event_packet_id::DATA_DESCRIPTOR_CHANGED)
     {
-        // drop packet if signal descriptors have not been changed
-        if (BaseObjectPtr::Equals(packet, getDataDescriptorChangedEventPacket(signalId)))
-            forwardPacket = false;
+        const auto valueDescriptorParam = packet.getParameters().get(event_packet_param::DATA_DESCRIPTOR);
+        const auto domainDescriptorParam = packet.getParameters().get(event_packet_param::DOMAIN_DATA_DESCRIPTOR);
+        const bool valueDescriptorChanged = valueDescriptorParam.assigned();
+        const bool domainDescriptorChanged = domainDescriptorParam.assigned();
+        const DataDescriptorPtr newValueDescriptor =
+            valueDescriptorParam != NullDataDescriptor() ? valueDescriptorParam : nullptr;
```

Comment:
This pattern is used very often so I would suggest having a standard function in openDAQ, perhaps called `validateDescriptor` or something like that.